tests/test_single_topology_stereo.py

```diff
@@ -0,0 +1,486 @@
+    # (ytz): discard burn in later
+    # burn_in_batches = num_batches//10
+    burn_in_batches = 0
+    frames = frames[:, burn_in_batches:, :, :]
+    # collect over all workers
+    frames = frames.reshape(-1, num_atoms, 3)
+
+    return frames
+
+
+def test_nblist_conversion():
+    mol = Chem.MolFromSmiles("CC1CC1C(C)(C)C")
+    bond_idxs = [[b.GetBeginAtomIdx(), b.GetEndAtomIdx()] for b in mol.GetBonds()]
+    nblist = geometry.bond_idxs_to_nblist(mol.GetNumAtoms(), bond_idxs)
+
+    expected = [[1], [0, 2, 3], [1, 3], [1, 2, 4], [3, 5, 6, 7], [4], [4], [4]]
+
+    np.testing.assert_array_equal(nblist, expected)
+
+
+def measure_chiral_volume(x0, x1, x2, x3):
+    """
+    Compute the normalized chiral volume given four points.
+
+    Parameters
+    ----------
+    np.array: (3,)
+        Center point
+
+    np.array: (3,)
+        First point
+
+    np.array: (3,)
+        Second point
+
+    np.array: (3,)
+        Third point
+
+    Returns
+    -------
+    float
+        A number between -1.0<x<1.0 denoting the normalized chirality
+
+    """
+    # compute vectors
+    v0 = x1 - x0
+    v1 = x2 - x0
+    v2 = x3 - x0
+
+    v0 = v0 / np.linalg.norm(v0)
+    v1 = v1 / np.linalg.norm(v1)
+    v2 = v2 / np.linalg.norm(v2)
+
+    return np.dot(np.cross(v0, v1), v2)
+
+
+def test_stereo_water_to_tetrahedral():
+    mol_a = Chem.MolFromMolBlock(
+        """
+  Mrv2202 05192218593D
+
+  3  2  0  0  0  0            999 V2000
+    0.5038    0.9940    0.3645 O   0  0  0  0  0  0  0  0  0  0  0  0
+    0.4342    0.1988    0.9333 H   0  0  0  0  0  0  0  0  0  0  0  0
+    0.5015    0.5419   -0.8126 H   0  0  0  0  0  0  0  0  0  0  0  0
+  1  3  1  0  0  0  0
+  1  2  1  0  0  0  0
+M  END
+$$$$""",
+        removeHs=False,
+    )
+
+    mol_b = Chem.MolFromMolBlock(
+        """
+  Mrv2202 05192219003D
+
+  5  4  0  0  0  0            999 V2000
+    0.5038    0.9940    0.3645 C   0  0  1  0  0  0  0  0  0  0  0  0
+    0.4342    0.1988    0.9333 H   0  0  0  0  0  0  0  0  0  0  0  0
+    0.5015    0.5419   -0.8126 F   0  0  0  0  0  0  0  0  0  0  0  0
+    1.6483    2.0245    0.3645 Cl  0  0  0  0  0  0  0  0  0  0  0  0
+   -0.5266    2.1385    0.3645 Br  0  0  0  0  0  0  0  0  0  0  0  0
+  1  3  1  0  0  0  0
+  1  2  1  0  0  0  0
+  1  4  1  0  0  0  0
+  1  5  1  0  0  0  0
+M  END
+$$$$""",
+        removeHs=False,
+    )
+
+    core = np.array([[0, 0], [1, 1], [2, 2]])
+    ff = Forcefield.load_from_file("smirnoff_1_1_0_sc.py")
+    s_top = SingleTopologyV2(mol_a, mol_b, core, ff)
+
+    x_a = utils.get_romol_conf(mol_a)
+    x_b = utils.get_romol_conf(mol_b)
+    x0 = s_top.combine_confs(x_a, x_b)
+
+    vol_cl = measure_chiral_volume(x0[0], x0[1], x0[2], x0[3])
+    vol_br = measure_chiral_volume(x0[0], x0[1], x0[2], x0[4])
+
+    assert vol_cl > 0 and vol_br < 0
+
+    idxs_and_params = s_top.generate_end_state_mol_a()
+    frames = simulate_idxs_and_params(idxs_and_params, x0)
```

Comment:
I don't see where we set a random seed here. Should we do that to ensure test determinism?

Reply:
hm lemme dig a bit deeper

Reply:
fixed